com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java

```diff
@@ -182,7 +184,17 @@ public void setExtraHeaders(List<String> val) {
 	public void setAcceptAllCertificates(boolean val) {
 		this.acceptAllCertificates = val;
 	}
-	
+	@Parameter(optional=true, 
```

Comment:
Since these are input attributes the setter should use the TupleAttribute interface. This makes the parameter of type attribute instead of rstring so that the caller does:
```
param postAttributes: id, message, when;
```
instead of
```
param postAttributes: "id", "message", "when";
```
The big advantage is that the SPL compiler will verify that the parameters exist and are of the correct type if TupleAttribute is typed. Thus the Java code operator does not have to implement any checking against the parameter.

I wasn't sure it TupleAttribute would support this case, a list of arbitrary typed tuples since TupleAttribute is normally used like:
```
@Parameter
public void setMyParam(TupleAttribute<Tuple,Integer> value) {
```
which is typed to be an Integer parameter, int32 in SPL.

However, I tried it out and it works, looking like this:
```
    private List<TupleAttribute<Tuple, ?>> postAtttributes;
    
    @Parameter
    public void setPostAtttributes(List<TupleAttribute<Tuple,?>> postAtttributes) {
    	this.postAtttributes = postAtttributes;
    }
```
The `?` makes the TupleAttribute untyped.

Then you can loop over the values of the attributes for an input tuple and the attribute names like:

```
    	for (TupleAttribute<Tuple, ?> postAttr : postAtttributes) {
    		Object attrValue = postAttr.getValue(tuple);
    		String attrName =  postAttr.getAttribute().getName();
    	}
```